### ISSUE TITLE
Poly: optimize single function vtables

### ISSUE THREAD

**Issue by skypjack** [skypjack/entt#689]
When vtables consist of a single function, we can optimize them and directly turn the vtable in a pointer to an erased function.
Thanks to @nikki93 for pointing this out!!

**Comment by nikki93** [comment]
Great! I wonder if it could actually be a template parameter -- like embed vtable yes / no or maximum size of embedded vtable. This way the client code can try it either way and decide based on empirical results. But the resulting implementation may not be worth it, so understandable to specialcase for single function also.

**Comment by skypjack** [comment]
I see the benefit of embed yes/no, but why would you want an _embed N_ if you don't have any control of what methods are actually embedded? It seems useless. It would be great to allow embedding some methods on choice, but I've no idea how difficult is this to do within EnTT poly. Let's see.

**Comment by nikki93** [comment]
Yeah just realized that. Maybe the best is being able to say which methods you'd want to embed for -- so eg. you can choose to do that for a draw method on drawables or an eval method for expressions, but keep other methods non embedded like a "getDescription" one.
